src/hotspot/share/gc/shenandoah/heuristics/shenandoahHeuristics.cpp

```diff
@@ -304,6 +436,155 @@ double ShenandoahHeuristics::time_since_last_gc() const {
 
 bool ShenandoahHeuristics::in_generation(ShenandoahHeapRegion* region) {
   return (_generation->generation_mode() == GLOBAL)
-      || (_generation->generation_mode() == YOUNG && region->affiliation() == YOUNG_GENERATION)
-      || (_generation->generation_mode() == OLD && region->affiliation() == OLD_GENERATION);
+    || (_generation->generation_mode() == YOUNG && region->affiliation() == YOUNG_GENERATION)
+    || (_generation->generation_mode() == OLD && region->affiliation() == OLD_GENERATION);
+}
+
+void ShenandoahHeuristics::prepare_for_old_collections() {
+  assert(_generation->generation_mode() == OLD, "This service only available for old-gc heuristics");
+  ShenandoahHeap* heap = ShenandoahHeap::heap();
+  uint free_regions = 0;
+  size_t cand_idx = 0;
+  size_t total_garbage = 0;
+  size_t num_regions = heap->num_regions();
+
+  RegionData* candidates = _region_data;
+  for (size_t i = 0; i < num_regions; i++) {
+    ShenandoahHeapRegion* region = heap->get_region(i);
+    if (!in_generation(region))
```

Comment:
Please always use { and } in if (and else) constructs.

Reply:
Ok.  Fixed.  (This code moved to shenadoahOldHeuristics.cpp)